### PR TITLE
OGM-1515 Support Java 8 LocalDate and LocalDateTime

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/OGM-1515

### CODE REVIEW

File: core/src/main/java/org/hibernate/ogm/type/impl/TimeGridTypeDescriptor.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.hibernate.ogm.model.spi.Tuple;
+import org.hibernate.ogm.type.descriptor.impl.BasicGridBinder;
+import org.hibernate.ogm.type.descriptor.impl.GridTypeDescriptor;
+import org.hibernate.ogm.type.descriptor.impl.GridValueBinder;
+import org.hibernate.ogm.type.descriptor.impl.GridValueExtractor;
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.JavaTypeDescriptor;
+
+/**
+ * Force to unwrap a {@link Time}
```
Comment (fax4ever) [inline core/src/main/java/org/hibernate/ogm/type/impl/TimeGridTypeDescriptor.java]: @DavideD I've used here the same strategy applied in `TimestampGridTypeDescriptor`

**Comment by gsmet** [comment]
@fax4ever minor gripe: could you please continue to use the full JIRA id in the name of the PR (it's your second PR in a row with this pattern)? We always included the `OGM-` prefix so I don't see any good reason to change that. Thanks!

**Comment by fax4ever** [comment]
@gsmet of course. I've forgotten to change it from the branch name.

File: core/src/test/java/org/hibernate/ogm/backendtck/type/datetime/LocalDateTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * Tests support for
+ * {@link java.time.LocalDate} and {@link java.time.LocalDateTime}
+ * as property types.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+@TestForIssue(jiraKey = "OGM-1515")
+@SkipByGridDialect(value = { MONGODB, INFINISPAN_REMOTE, NEO4J_REMOTE }, comment = "Dialects need to handle java.time.LocalDate and java.time.LocalDateTime as property types")
```
[the remark this answers is not in the document]
Comment (fax4ever) [reply core/src/test/java/org/hibernate/ogm/backendtck/type/datetime/LocalDateTest.java]: I think that the issue is scoped on just the core. Then, applying it, the expectation should be that `LocalDate` and `LocalDateTime` are be supported by all dialects that support these classes natively.

[the remark this answers is not in the document]
Comment (fax4ever) [reply core/src/test/java/org/hibernate/ogm/backendtck/type/datetime/LocalDateTest.java]: Thus my idea is to going to open other issues for enabling support for the remaining dialects. Thus avoiding putting everything in the same issue. But we can also do the different way. as you wish: P

[the remark this answers is not in the document]
Comment (DavideD) [reply core/src/test/java/org/hibernate/ogm/backendtck/type/datetime/LocalDateTest.java]: Unless there are complications that make the task difficult, I would prefer if we fix the issue for all the dialects right now. JIRAs are not written in stone, the scope cant change when we understand what we need to do for a task and we can edit them if needed.

[the remark this answers is not in the document]
Comment (DavideD) [reply core/src/test/java/org/hibernate/ogm/backendtck/type/datetime/LocalDateTest.java]: This way it also easier to keep dialects aligned on features

[the remark this answers is not in the document]
Comment (fax4ever) [reply core/src/test/java/org/hibernate/ogm/backendtck/type/datetime/LocalDateTest.java]: It's OK. I'll add support for every core dialects in subsequent commits.

**Comment by fax4ever** [comment]
PR is a mark a `Preview`, because we have to support `LocalDate` types for all core dialects.

**Comment by fax4ever** [comment]
I've implemented the relative internal conversions on dialects and rebased the commits in a more natural way.

**Comment by DavideD** [comment]
@fax4ever  This looks good and I'm going to merge it but could you create a follow up issue where we test how the values appear in the db natively? We want to make sure that in MongoDb and Neo4j the string is stored in the format we expect (we already have similar tests under the `mapping` package). If possible even in Infinspan.

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-ogm/pull/1065

Thanks

**Comment by fax4ever** [comment]
Thank you!

**Comment by fax4ever** [comment]
Yes. I'm going to open the follow up issue.

**Pull request closed without merging**